rasterio/_base.pyx

```diff
@@ -330,23 +330,67 @@ cdef class DatasetReader(object):
             return flags
 
     def block_windows(self, bidx=0):
-        """Returns an iterator over a band's block windows and their
-        indexes.
+        """Returns an iterator over a band's blocks and their corresponding
+        windows.  Produces tuples like ``(block, window)``.  The primary use
+        of this method is to obtain windows to pass to `read()` for highly
+        efficient access to raster block data.
+
+        The positional parameter `bidx` takes the index (starting at 1) of the
+        desired band.  This iterator yields blocks "left to right" and "top to
+        bottom" and is similar to Python's ``enumerate()`` in that the first
+        element is the block index and the second is the dataset window.
+
+        Blocks describe how pixels are grouped within each band and provide a
+        mechanism for efficient I/O.  A window is a range of pixels within a
+        single band defined by row start, row stop, column start, and column
+        stop.  For example, ``((0, 2), (0, 2))`` defines a ``2 x 2`` block at
+        the upper left corner of a raster band.  Blocks are also tuples like
+        ``(0, 0)``, which would be a band's upper left block.
+
+        Raster I/O is performed at the block level, so accessing a window
+        spanning multiple rows in a striped raster requires reading each row.
+        Accessing a ``2 x 2`` window at the center of a ``1800 x 3600`` image
+        requires reading 2 rows, or 7200 pixels just to get the target 4.  The
+        same image with internal ``256 x 256`` tiles would require reading at
+        least 1 tile (if the window entire window falls within a single tile)
+        and at most 4 tiles, or at least 512 pixels and at most 2048.
+
```

Comment:
@geowurster is "striped raster" and "raster without blocks" the same thing?  If so, the latter might be a little more clear in this context.  

Is usage of the term 'tile' important here?  Seems like this would read more consistently if you replaced 'tiles' with 'blocks'.